### PR TITLE
[IMP] saas-15: Subscription flow

### PR DESCRIPTION
**Pull request by tbe-odoo**
Description of the issue/feature this PR addresses:
- Checking the validity of credit cards.
- Ability to change the credit card assigned to a subscription.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/payment/models/payment_acquirer.py [hunk 1]
```diff
@@ -684,6 +684,16 @@ class PaymentToken(models.Model):
     acquirer_ref = fields.Char('Acquirer Ref.', required=True)
     active = fields.Boolean('Active', default=True)
     payment_ids = fields.One2many('payment.transaction', 'payment_token_id', 'Payment Transactions')
+    verification_status = fields.Selection([
+        ('none', 'Unverified'),
```
Comment (dbo-odoo) [inline addons/payment/models/payment_acquirer.py]: unverified instead of none

File: addons/payment_ogone/models/payment.py [hunk 2]
```diff
@@ -409,6 +419,7 @@ def _ogone_s2s_validate_tree(self, tree, tries=2):
                     'name': tree.get('CARDNO'),
                 })
                 self.write({'payment_token_id': pm.id})
+            self.payment_token_id.verification_status = 'verified'
```
Comment (dbo-odoo) [inline addons/payment_ogone/models/payment.py]: should be a boolean instead, set to true when a payment is successful

File: addons/payment_ogone/models/payment.py [hunk 3]
```diff
@@ -529,3 +543,154 @@ def ogone_create(self, values):
                 'name': 'XXXXXXXXXXXX%s - %s' % (values['cc_number'][-4:], values['cc_holder_name'])
             }
         return {}
+
+    def ogone_validate(self):
+        self.verification_status = 'pending'
```
Comment (antonylesuisse) [inline addons/payment_ogone/models/payment.py]: env['payment.transaction'].create({}).send().refund()

**Comment by KangOl** [comment]
Should target `master`

**Comment by tbe-odoo** [comment]
@KangOl my local environment is running on saas-15 (working on a copy of our prod) that's why I'm working on Saas-15, but I'll rebase this to target master ;)

**Comment by KangOl** [comment]
Why closed?

**Pull request closed without merging**